Cql/Cql.Compiler/TypeManager.cs

```diff
@@ -120,10 +121,10 @@ public TypeManager(TypeResolver resolver, string assemblyName = "Tuples", string
                 {
                     Type? sourceType = null;
                     if (propertyExpression.source != null)
-                        sourceType = TypeFor(propertyExpression.source!, context);
+                        sourceType = TypeFor(propertyExpression.source!, ctx.Deeper(propertyExpression.source));

```

Comment:
That looks like a real change in the moment when ctx.Deeper is called - is this on purpose?

Reply:
Yes, this is on purpose. Exceptions thrown in TypeFor must be thrown such that the `source` element is dumped as part of the exception message.